src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1202,43 +1157,49 @@ public List<String> getInactiveConsumerIds(Instant lastCheckedInRetention, Insta
      *     Org namespace to search
      * @return Consumer that matches the given
      */
-    @SuppressWarnings("unchecked")
     @Transactional
-    public CandlepinQuery<Consumer> getHypervisorsBulk(Iterable<String> hypervisorIds, String ownerId) {
+    public List<Consumer> getHypervisorsBulk(Collection<String> hypervisorIds, String ownerId) {
         if (hypervisorIds == null || !hypervisorIds.iterator().hasNext()) {
-            return this.cpQueryFactory.<Consumer>buildQuery();
+            return new ArrayList<>();
         }
 
-        DetachedCriteria criteria = DetachedCriteria.forClass(Consumer.class)
-            .createAlias("hypervisorId", "hvsr")
-            .add(Restrictions.eq("ownerId", ownerId))
-            .add(this.getHypervisorIdRestriction(hypervisorIds))
-            .addOrder(org.hibernate.criterion.Order.asc("hvsr.hypervisorId"))
-            .setFetchMode("type", FetchMode.SELECT);
+        List<String> lowerCaseHypervisorIds = hypervisorIds.stream()
```

Comment:
As someone who enjoys streams, I like this solution; but I don't like that it means we're generating a new list just to throw it out again as we start partitioning.

Let me think on this for a bit to figure out a clean way to build a stream that we can throw at a partitioner without needing to recollect it first.

Reply:
Looks like `Stream.iterator` is also lazy, so we can use the magic of functional interfaces to pass that along the stream without first recollecting it:

```
Stream<String> hypervisorIds.stream()
  .distinct()
  ...

for (List<String> block : this.partition(lowerCaseHypervisorIds::iterator)) {
```

Reply:
done